core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -62,15 +62,44 @@ static PageStore create(PageStoreOptions options) {
     return pageStore;
   }
 
+  /**
+   * Writes a new temporary page from a source channel to the store.
+   *
+   * @param pageId page identifier
+   * @param page page data
+   * @throws ResourceExhaustedException when there is not enough space found on disk
+   * @throws IOException when the store fails to write this page
+   */
+  default void putTemporary(PageId pageId,
```

Comment:
Do you think it makes sense to move the capability of adding temporary pages to a child interface of `PageStore`, say `BlockPageStore` that's specifically for use in the block worker? Or are there any potential use cases for the client-side local cache to use temporary pages as well?

Reply:
We also need `commitTemporary` (or at least `movePage`) in the said `BlockPageStore` to be able to implement `commitBlock` in `PagedBlockStore`.